src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy

```diff
@@ -279,6 +332,200 @@ class Server42RttSpec extends HealthCheckSpecification {
+                    [flowid   : flow.flowId,
+                     direction: "reverse"]).dps
+            assert newStatsDataReverse.size() > statsDataReverse.size()
+        }
+        otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                [flowid   : flow.flowId,
+                 direction: "forward"]).dps.size() == statsDataForward.size()
+
+        when: "Synchronize the flow"
+        with(northbound.synchronizeFlow(flow.flowId)) { !it.rerouted }
+
+        then: "Missing ingress server42 rule is reinstalled on the src switch"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            assert northbound.validateSwitch(switchPair.src.dpId).rules.missing.empty
+            assert northbound.getSwitchRules(switchPair.src.dpId).flowEntries.findAll {
+                new Cookie(it.cookie).getType() == CookieType.SERVER_42_INGRESS
+            }*.cookie.size() == 1
+        }
+
+        then: "server42 stats for forward direction are available again"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT + WAIT_OFFSET, 1) {
+            assert otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : flow.flowId,
+                     direction: "forward"]).dps.size() > statsDataForward.size()
+            assert otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : flow.flowId,
+                     direction: "reverse"]).dps.size() > newStatsDataReverse.size()
+        }
+
+        cleanup: "Revert system to original state"
+        revertToOrigin([flow], flowRttFeatureStartState, initialSwitchRtt)
+    }
+
+    @Tidy
+    def "Able to swapEndpoint for a flow with enabled server42 on it"() {
+        given: "Two switch pairs with different src switches and the same dst switch"
+        def switchIdsConnectedToS42 = topology.getActiveServer42Switches()*.dpId
+        SwitchPair fl2SwPair = null
+        SwitchPair fl1SwPair = topologyHelper.switchPairs.collectMany { [it, it.reversed] }.find { firstSwP ->
+            def firstOk =
+                    firstSwP.src.dpId in switchIdsConnectedToS42 && !switchIdsConnectedToS42.contains(firstSwP.dst.dpId)
+            fl2SwPair = topologyHelper.switchPairs.collectMany { [it, it.reversed] }.find { secondSwP ->
+                secondSwP.dst.dpId == firstSwP.dst.dpId && secondSwP.src.dpId != firstSwP.src.dpId &&
+                        !switchIdsConnectedToS42.contains(secondSwP.src.dpId)
+            }
+            firstOk && fl2SwPair
+        }
+        assumeTrue("Required switch pairs were not found in the given topology",
+                fl1SwPair.asBoolean() && fl2SwPair.asBoolean())
+
+        and: "server42 is enabled on the src sw of the first switch pair"
+        def flowRttFeatureStartState = changeFlowRttToggle(true)
+        def initialSwitchRtt = [].collectEntries { [fl1SwPair.src, changeFlowRttSwitch(fl1SwPair.src, true)] }
```

Comment:
I'm not sure what this line is supposed to do, but right now it is incorrect, you are trying to iterate over an empty array

Reply:
ops, fixed.
thx.